File: CSharp-01-Fundamentals/04. Console-In-and-Out/Homework/04. Console-In-and-Out/P06. Quadratic Equation/P06. Quadratic Equation.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
/********************************************************************************************
    Quadratic Equation

    Description

    Write a program that reads the coefficients a, b and c of a quadratic equation ax2 + bx + c = 0 and solves it (prints its real roots).

    Input

    On the first three lines, you will receive the coefficients a, b, and c, each on a separate line in the same order
    Output

    If two different real roots exist, print them on two separate lines
    Print the smaller root on the first line
    If only one real root exists, print it on the only output line
    If no real root exists, print the string "no real roots"
    The roots, should they exist, must be printed with precision exactly two digits after the floating point
    Constraints

    The input will always consist of valid real numbers in the range [-1000, 1000] and will follow the described format
    Time limit: 0.1s
    Memory limit: 8MB
    Sample tests

Input	Output
2
5
-3	-3,00
0,50

-1
3
0	0,00
3,00

-0,5
4
-8	4,00

5
2
8	no real roots

*********************************************************************************************/
namespace P06.Quadratic_Equation
{
    class QuadraticEquation
    {
        static void Main(string[] args)
        {
            float a = float.Parse(Console.ReadLine());
            float b = float.Parse(Console.ReadLine());
            float c = float.Parse(Console.ReadLine());

            double D = Math.Pow(b, 2) - (4 * a * c);
            double? x1;
            double? x2;

            if (D < 0)
            {
                x1 = null;
                x2 = null;
            }
            else if (D == 0)
            {
                x1 = (-1) * b / (2 * a);
                x2 = null;
            } else
            {
                D = Math.Sqrt(D);
                x1 = ((-1) * b + D) / (2 * a);
                x2 = ((-1) * b - D) / (2 * a);
            }

            if (x1 == null && x2 == null)
            {
                Console.WriteLine("no real roots");
            }
            else if (x1 != null && x2 == null)
            {
                Console.WriteLine("{0:#0.00}", x1);
            }
            else
            {
                double xMin = Math.Min((double)x1, (double)x2);
                double xMax = Math.Max((double)x1, (double)x2);
                Console.WriteLine("{0:#0.00}", xMin);
                Console.WriteLine("{0:#0.00}", xMax);
            }            
        }
    }
}

File: My-Homeworks-ExamPreparations-And-Excersizes/JavaScript Fundamentals/05. Conditional Statements/Homework/P06. Quadratic equation/P06. Quadratic equation.js
/**
 * Created by Admin on 23.6.2016 г..
 */
/*
 Quadratic equation

 Description

 Write a script that reads the coefficients  a ,  b  and  c  of a quadratic equation  ax2 + bx + c = 0
 and solves it (prints its real roots). Calculates and prints its real roots.

 Note: Quadratic equations may have 0, 1 or 2 real roots.

 Input
 •The input will consist of an array containing three values -  a ,  b  and  c  represented as strings

 Output
 •The output should be a single line containing the real roots (see sample tests) ◦
 Print numbers with two digits of precision after the floating point
 If there are two roots then  x1 < x2


 Constraints
 •Time limit: 0.2s
 •Memory limit: 16MB

 Sample tests
 Input
 Output
 2
 5
 -3 x1=-3.00; x2=0.50
 -1
 3
 0 x1=0.00; x2=3.00
 -0.5
 4
 -8 x1=x2=4.00
 5
 2
 8 no real roots
 0.2
 9.572
 0.2 x1=-47.84; x2=-0.02
 */

// var input = [2, 5, -3, -1, 3, 0, -0.5, 4, -8, 5, 2, 8, 0.2, 9.572, 0.2];
// solve(input);

function solve(args) {
    // var argsLen = args.length;
    // for(var i = 2; i < argsLen; i = i + 3){
    //     var a = args[i - 2];
    //     var b = args[i - 1];
    //     var c = args[i];
    //
    //     quadraticEquation(a, b, c);
    // }

    var a = args[0];
    var b = args[1];
    var c = args[2];

    quadraticEquation(a, b, c);

    function quadraticEquation(a, b, c) {
        a = parseFloat(a);
        b = parseFloat(b);
        c = parseFloat(c);
        var D = 0.0
        var x1 = 0.0;
        var x2 = 0.0;

        D = (b * b) - (4 * a * c);

        if(D === 0) {
            x1 = -b / (2 * a);
            console.log("x1=x2=" + x1.toFixed(2));
        }
        else if(D > 0) {
            x1 = (-b - Math.sqrt(D)) / (2 * a);
            x2 = (-b + Math.sqrt(D)) / (2 * a);
            console.log("x1=" + x1.toFixed(2) + "; x2=" + x2.toFixed(2));
        }
        else {
            console.log("no real roots");
        }
    }

}



